src/strategies/BeefyAeroStrategy.sol

```diff
@@ -0,0 +1,313 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity =0.8.16;
+
+import {FixedPointMathLib} from "solmate/src/utils/FixedPointMathLib.sol";
+import {SafeTransferLib} from "solmate/src/utils/SafeTransferLib.sol";
+
+import {Math} from "@openzeppelin/contracts/utils/math/Math.sol";
+import {ERC20} from "solmate/src/tokens/ERC20.sol";
+
+import {IAeroRouter, IAeroPool} from "src/interfaces/IAerodrome.sol";
+import {IBeefyVault} from "src/interfaces/Beefy.sol";
+
+import {AccessStrategy} from "src/strategies/AccessStrategy.sol";
+
+import {AffineVault} from "src/vaults/Vault.sol";
+import {StrategyVault} from "src/vaults/locked/StrategyVault.sol";
+
+contract BeefyAeroStrategy is AccessStrategy {
+    using FixedPointMathLib for uint256;
+    using SafeTransferLib for ERC20;
+
+    // beefy vault
+    IBeefyVault public immutable beefy;
+    // const max bps
+    uint256 public constant MAX_BPS = 10_000;
+    uint256 public defaultSlippageBps;
+
+    ERC20 public immutable token0;
+    ERC20 public immutable token1;
+
+    IAeroRouter public immutable aeroRouter;
+    IAeroPool public immutable lpToken;
```

Comment:
We should also set the factory here, instead of doing a call to find the factory in all of the other functions.